actors/builtin/miner/miner_state.go

```diff
@@ -208,6 +209,10 @@ func (st *State) GetStorageWeightDescForSector(store adt.Store, sectorNo abi.Sec
 	return asStorageWeightDesc(st.Info.SectorSize, sectorInfo), nil
 }
 
+func (st *State) inChallengeWindow(rt Runtime) bool {
+	return rt.CurrEpoch() > st.PoStState.ProvingPeriodStart // TODO: maybe also a few blocks beforehand?
```

Comment:
nah... just lengthen the challengeePeriod

Reply:
oh, i mean that you could hypothetically know the randomness at the challenge issuance a block or two in advance if youre a miner, and you could then submit faults just before that to cheat your post